src/backend/OpenCL/CHIPBackendOpenCL.cc

```diff
@@ -1291,6 +1366,401 @@ std::shared_ptr<chipstar::Event> CHIPQueueOpenCL::enqueueBarrierImpl(
   return Event;
 }
 
+/********************************************************************************/
+
+chipstar::GraphNative *CHIPQueueOpenCL::createNativeGraph() {
+  // should not raise an error if we fail to create a graph,
+  // because there is a fallback solution
+  CHIPContextOpenCL *Ctx = (CHIPContextOpenCL *)ChipContext_;
+  if (!Ctx->supportsCommandBuffers())
+    return nullptr;
+
+  cl_command_queue CQ = ClQueue_->get();
+  int err = CL_SUCCESS;
+  cl_command_buffer_khr Res =
```

Comment:
Add a check somewhere that we are targeting only one device, if using the single-device CB.